crypto/s2n_cipher.c

```diff
@@ -23,6 +26,11 @@ int s2n_session_key_alloc(struct s2n_session_key *key)
 {
     eq_check(key->evp_cipher_ctx, NULL);
     notnull_check(key->evp_cipher_ctx = EVP_CIPHER_CTX_new());
+#if defined(S2N_CIPHER_AEAD_API_AVAILABLE)
```

Comment:
Is there anyway to tell which one is actually about to be used to avoid allocating both?

Reply:
That seems to require en API change to `s2n_session_key_alloc`, as well as assuming that when allocating the `s2n_session_key` object the negotiated cipher suite is known.